homeassistant/components/prometheus/sensor.py

```diff
@@ -0,0 +1,115 @@
+"""Prometheus Sensor component."""
+from datetime import timedelta
+import logging
+from typing import Union
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA as SENSOR_PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_UNIT_OF_MEASUREMENT,
+    CONF_URL,
+    STATE_UNKNOWN,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+DEFAULT_URL = "http://localhost:9090"
+CONF_QUERIES = "queries"
+CONF_EXPR = "expr"
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+_LOGGER = logging.getLogger(__name__)
+
+_QUERY_SCHEMA = vol.Schema(
+    {
+        vol.Required(
+            CONF_NAME,
+        ): cv.string,
+        vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,
+        vol.Required(CONF_EXPR): cv.string,
+    }
+)
+
+PLATFORM_SCHEMA = SENSOR_PLATFORM_SCHEMA.extend(
```

Comment:
I don't know what's idiomatic here, but I believe most integrations create one entity per each defined `sensor`, while in this case a single sensor definition will might result in several sensor entities created. Flattening the configuration structure might make it slightly more straightforward (even if a bit more verbose).

What I mean is something like this:
```
sensor:
  - platform: prometheus
    name: entropy
    query: min(node_entropy_available_bits{instance="localhost:9100"})
  - platform: prometheus
    name: free disk space
    url: http://localhost:9090
    query: min(node_filesystem_free_bytes{instance="localhost:9100", mountpoint="/"})
    unit_of_measurement: Bytes
```

Anyway, hopefully one of the Home Assistant core mainteners will weigh in and provide guidance here.

Reply:
Then you'd have to pass the `url` over and over again. The way I did it is basically how the influxdb sensor does it, so I think this is fine.

I guess you can still create multiple prometheus platforms for different prometheus instances though.

Reply:
We no longer allow new integrations to use a platform config section or old integrations to do changes to the platform config section. See https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md.

Please refactor this integration to have the config schema in the base integration module and the config yaml section under the top level integration domain key.